deployer/src/persistence/mod.rs

```diff
@@ -604,75 +604,126 @@ mod tests {
+            address: Some(SocketAddr::new(Ipv4Addr::LOCALHOST.into(), 9876)),
+            is_next: true,
+        };
+
+        for deployment in [
+            &deployment_other,
+            &deployment_crashed,
+            &deployment_stopped,
+            &deployment_running,
+        ] {
+            p.insert_deployment(deployment.clone()).await.unwrap();
+        }
+
+        let actual = p.get_deployments(&service_id).await.unwrap();
+        let expected = vec![deployment_stopped, deployment_crashed, deployment_running];
+
+        assert_eq!(actual, expected, "deployments should be sorted by time");
+    }
+
     // Test that we are correctly cleaning up any stale / unexpected states for a deployment
     // The reason this does not clean up two (or more) running states for a single deployment is because
     // it should theoretically be impossible for a service to have two deployments in the running state.
-    // And even if a service where to have this, then the start ups of these deployments (more specifically
+    // And even if a service were to have this, then the start ups of these deployments (more specifically
     // the last deployment that is starting up) will stop all the deployments correctly.
     #[tokio::test(flavor = "multi_thread")]
     async fn cleanup_invalid_states() {
         let (p, _) = Persistence::new_in_memory().await;
 
         let service_id = add_service(&p.pool).await.unwrap();
-
-        let queued_id = Uuid::new_v4();
-        let building_id = Uuid::new_v4();
-        let built_id = Uuid::new_v4();
-        let loading_id = Uuid::new_v4();
+        let time = Utc::now();
 
         let deployment_crashed = Deployment {
             id: Uuid::new_v4(),
             service_id,
             state: State::Crashed,
-            last_update: Utc::now(),
+            last_update: time,
             address: None,
             is_next: false,
         };
         let deployment_stopped = Deployment {
             id: Uuid::new_v4(),
             service_id,
             state: State::Stopped,
-            last_update: Utc::now(),
+            last_update: time.checked_add_signed(Duration::seconds(1)).unwrap(),
             address: None,
             is_next: false,
         };
         let deployment_running = Deployment {
             id: Uuid::new_v4(),
             service_id,
             state: State::Running,
-            last_update: Utc::now(),
+            last_update: time.checked_add_signed(Duration::seconds(2)).unwrap(),
             address: Some(SocketAddr::new(Ipv4Addr::LOCALHOST.into(), 9876)),
             is_next: false,
         };
         let deployment_queued = Deployment {
-            id: queued_id,
+            id: Uuid::new_v4(),
             service_id,
             state: State::Queued,
-            last_update: Utc::now(),
+            last_update: time.checked_add_signed(Duration::seconds(3)).unwrap(),
```

Comment:
The next one here is also 3 seconds, maybe this will fail sporadically? :thinking: